### PR TITLE
Add ConvertManifestToPackageJson editor

### PR DESCRIPTION
**Pull request by cdupuis**

### CODE REVIEW

File: .atomist/editors/ConvertExistingProjectToGenerator.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { EditProject } from "@atomist/rug/operations/ProjectEditor";
 import { Pattern } from "@atomist/rug/operations/RugOperation";
 
 @Editor("ConvertExistingProjectToGenerator",
-    "converts and existing project to a Rug archive project with a basic Generator")
+    "converts an existing project to a Rug archive project with a basic Generator")
```
Comment (ddgenome) [inline .atomist/editors/ConvertExistingProjectToGenerator.ts]: Thanks

File: .atomist/editors/ConvertManifestToPackageJson.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+import { Project } from "@atomist/rug/model/Project";
+import { Editor, Parameter, Tags } from "@atomist/rug/operations/Decorators";
+import { EditProject } from "@atomist/rug/operations/ProjectEditor";
+import { Pattern } from "@atomist/rug/operations/RugOperation";
+
+import * as yaml from "js-yaml";
+
+const packageJsonName = ".atomist/package.json";
+const manifestYamlName = ".atomist/manifest.yml";
```
Comment (ddgenome) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: These could be moved inside the class.

File: .atomist/editors/ConvertManifestToPackageJson.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            project.findFile(packageJsonName).setContent(newPackageJsonContent);
+            project.deleteFile(manifestYamlName);
+        }
+    }
+
+    private loadPackageJson(project: Project): any {
+        if (project.fileExists(packageJsonName)) {
+            return JSON.parse(project.findFile(packageJsonName).content);
+        } else {
+            project.addFile(packageJsonName, "{}");
```
Comment (ddgenome) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: Might be better to copy the one from this archive to give a more reasonable starting point.

File: .atomist/editors/ConvertManifestToPackageJson.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            }
+
+            atomist.requires = manifestYaml.requires;
+
+            // Add in dependencies
+            if (manifestYaml.dependencies != null) {
+                atomist.dependencies = {};
+                for (const d of manifestYaml.dependencies) {
+                    const parts = (d as string).split(":");
+                    atomist.dependencies[`${parts[0]}:${parts[1]}`] = parts[2];
```
Comment (ddgenome) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: Isn't version optional for a dependency in the manifest? Can our package.json parser deal with an empty value for a dependency, i.e., if `parts[2]` is null?

File: .atomist/editors/ConvertManifestToPackageJson.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            if (newPackageJson.atomist == null) {
+                newPackageJson.atomist = atomist;
+            } else {
+                atomist = newPackageJson.atomist;
+            }
+
+            atomist.requires = manifestYaml.requires;
+
+            // Add in dependencies
+            if (manifestYaml.dependencies != null) {
```
Comment (ddgenome) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: Why isn't TSLint complaining that this isn't `!==`?

File: .atomist/editors/ConvertManifestToPackageJson.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                    atomist.dependencies[`${parts[0]}:${parts[1]}`] = parts[2];
+                }
+            }
+
+            // Add in extensions
+            if (manifestYaml.extensions != null) {
+                atomist.extensions = {};
+                for (const d of manifestYaml.extensions) {
+                    const parts = (d as string).split(":");
+                    atomist.extensions[`${parts[0]}:${parts[1]}`] = parts[2];
```
Comment (ddgenome) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: Same issue around null `parts[2]`.

File: .atomist/editors/ConvertManifestToPackageJson.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            if (manifestYaml.dependencies != null) {
+                atomist.dependencies = {};
+                for (const d of manifestYaml.dependencies) {
+                    const parts = (d as string).split(":");
+                    atomist.dependencies[`${parts[0]}:${parts[1]}`] = parts[2];
+                }
+            }
+
+            // Add in extensions
+            if (manifestYaml.extensions != null) {
```
Comment (ddgenome) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: Perhaps this allows extensions to be null or undefined?

File: .atomist/tests/project/ConvertManifestToPackageJson.feature [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  Use an editor to convert an existing manfiest.yml to a package.json
+  honoring existing content in package.json. The editor create a new 
+  package.json file if non exists.
+
+  Scenario: Do not convert package.json if no manifest.yml exists
+    Given an empty project
+    When the ConvertManifestToPackageJson is run
+    Then no changes were made
+
+Scenario: Merge manifest.yml into existing package.json
```
Comment (ddgenome) [inline .atomist/tests/project/ConvertManifestToPackageJson.feature]: Indentation

File: .atomist/tests/project/ConvertManifestToPackageJson.feature [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    Given a manifest.yml and package.json
+    When the ConvertManifestToPackageJson is run
+    Then the manifest.yml got merged into package.json
+    Then the package.json has previous entries
+    Then the manifest.yml got deleted
+
+Scenario: Convert manifest.yml into new package.json
```
Comment (ddgenome) [inline .atomist/tests/project/ConvertManifestToPackageJson.feature]: Indentation

File: .atomist/tests/project/ConvertManifestToPackageJson.feature [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  honoring existing content in package.json. The editor create a new 
+  package.json file if non exists.
+
+  Scenario: Do not convert package.json if no manifest.yml exists
+    Given an empty project
+    When the ConvertManifestToPackageJson is run
+    Then no changes were made
+
+Scenario: Merge manifest.yml into existing package.json
+    Given a manifest.yml and package.json
```
Comment (ddgenome) [inline .atomist/tests/project/ConvertManifestToPackageJson.feature]: You could break this into two `Given` statements and simplify the implementations in the Steps.ts file.

File: .atomist/editors/ConvertManifestToPackageJson.ts (hunk 2 above)
Comment (cdupuis) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: ok

File: .atomist/editors/ConvertManifestToPackageJson.ts (hunk 5 above)
Comment (cdupuis) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: No idea.

File: .atomist/editors/ConvertManifestToPackageJson.ts (hunk 4 above)
Comment (cdupuis) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: No, it is not optional.

File: .atomist/editors/ConvertManifestToPackageJson.ts (hunk 7 above)
Comment (cdupuis) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: Not sure I understand?

File: .atomist/editors/ConvertManifestToPackageJson.ts (hunk 3 above)
Comment (cdupuis) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: Ok

Comment (cdupuis) [inline .atomist/editors/ConvertManifestToPackageJson.ts]: We may want to delete the js-yaml stuff out of it though.

File: .atomist/tests/project/ConvertManifestToPackageJson.feature (hunk 10 above)
Comment (cdupuis) [inline .atomist/tests/project/ConvertManifestToPackageJson.feature]: Ok

**Comment by ddgenome** [comment]
Is it not an issue if a dependency/extension version is not specified in the manifest.yml/package.json?

Change log entry?

**Comment by cdupuis** [comment]
Added changelog entry. 

> Is it not an issue if a dependency/extension version is not specified in the manifest.yml/package.json?

If you don't have a version for your dependencies/extensions in your `manifest.yml` the CLI will barf.

**Pull request merged**